### PR TITLE
Managed resource guard #35

### PR DESCRIPTION
**Pull request by pepov**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
Adds a label to all created resources to be able to detect if a resource was managed by the cli or not. Provide a survey for interactive terminals to allow deciding whether a resource should be managed or not if it's not already, otherwise skip.

Make cert-manager install/uninstall simpler so that it only checks for an existing namespace, since all the other resource will be checked by the new guard anyways.

### Why?
The `uninstall -a` command removes cert-manager even if it was created externally, which we want to avoid generally.

### Checklist

- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)
- [ ] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: internal/k8s/label.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	lm.setDesiredLabels(desired)
+	return false, nil
+}
+
+func (lm *labelManager) CheckLabelsBeforeDelete(actual *unstructured.Unstructured) (bool, error) {
+	labels := actual.GetLabels()
+	if _, ok := labels["backyards.banzaicloud.io/cli-version"]; !ok {
+		if lm.deleteAll {
+			return false, nil
+		} else if !lm.skipAll {
```
Comment (tothandras) [inline internal/k8s/label.go]: `else` is unnecessary

File: internal/k8s/label.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+func (lm *labelManager) CheckLabelsBeforeUpdate(actual, desired *unstructured.Unstructured) (bool, error) {
+	labels := actual.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	if _, ok := labels["backyards.banzaicloud.io/cli-version"]; !ok {
+		if lm.manageAll {
+			lm.setDesiredLabels(desired)
+			return false, nil
+		} else if !lm.skipAll {
```
Comment (tothandras) [inline internal/k8s/label.go]: same

File: internal/k8s/label.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	lm.setDesiredLabels(desired)
+	return false, nil
+}
+
+func (lm *labelManager) setDesiredLabels(desired *unstructured.Unstructured) {
+	labels := desired.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	if _, ok := labels["backyards.banzaicloud.io/cli-version"]; !ok {
```
Comment (tothandras) [inline internal/k8s/label.go]: we should make a `const` for `"backyards.banzaicloud.io/cli-version"`

File: internal/k8s/label.go (hunk 1 above)
Comment (pepov) [reply internal/k8s/label.go]: I don't see why

File: internal/k8s/label.go (hunk 3 above)
Comment (pepov) [reply internal/k8s/label.go]: sure

File: internal/k8s/label.go (hunk 1 above)
Comment (waynz0r) [reply internal/k8s/label.go]: because of the `return`

**Pull request merged**